main.js

```diff
@@ -0,0 +1,310 @@
+  getLocationCoordinates(input, true)
+})
+
+//used in geolocation: gets a location name from the location address data to pass to the getLocationCoordinates function to get weather data
+var parseLocation = function(data) {
+  //iterates through location address data to find the item that gives the locality or sublocality name to pass to the getLocationCoordinates function. If no name, alerts an error and hides the spinner
+  var locationName
+  for (i = 0; i < data.results.length; i++) {
+    for (j = 0; j < data.results[i].types.length; j++) {
+      if (data.results[i].types[j] == "locality" || data.results[i].types[j] == "sublocality") {
+        locationName = data.results[i].formatted_address;
+        break;
+      }
+    }
+  }
+  if (!locationName) {
+    $('.spinner').addClass('hide')
+    alert("Geolocation error, please try again.")
+  }
+  $('#search-query').val(locationName)
+  getLocationCoordinates(locationName, false)
+}
+//pass search query to a Google maps API to get full name, latitude, and longitude then pass latitude and longitude to the openweather API to get weather data
+var getWeather = function(data, isDefault) {
+  $('.weekly-forecast').removeClass('hide')
+  $('.daily-forecast').removeClass('hide')
+  $('.spinner').addClass('hide')
+  //if we're getting weather for the default location, we set the default location name
+  if (isDefault) {
+    defaultLocationName = data.results[0].formatted_address
+  }
+  var locationName = data.results[0].formatted_address
+  //checks whether a "normal search" is for the default location - if so we hide the active default location button and show the disabled default location button
+  if (locationName == defaultLocationName) {
+    $('.not-default').addClass('hide');
+    $('.is-default').removeClass('hide');
+  } else {
+    $('.not-default').removeClass('hide');
+    $('.is-default').addClass('hide');
+  }
+  //remove the ', USA' from cities in the United States to improve display
+  if (locationName.search(/,\sUSA/) != -1) {
+    locationName = locationName.slice(0, locationName.search(/,\sUSA/));
+  }
+  locationLatitude = data.results[0].geometry.location.lat;
+  locationLongitude = data.results[0].geometry.location.lng;
+  getDailyWeather(locationName, locationLatitude, locationLongitude);
+  getWeeklyWeather(locationLatitude, locationLongitude);
+}
+
+var renderDailyForecast = function() {
+  var source = $('#today-forecast-template').html()
+  var template = Handlebars.compile(source)
+  var newHtml = template(today)
+  $('.daily-forecast').html(newHtml)
+}
+
+var renderWeeklyForecast = function() {
+  $('.weekly-forecast').html("")
+  var source = $('#five-day-forecast-template').html()
+  var template = Handlebars.compile(source)
+  for (i = 0; i < 5; i++) {
+    newHtml = template(weather[i])
+    $('.weekly-forecast').append(newHtml)
+  }
+}
+
+//generate the today object with current weather data
+var currentWeather = function(data, location) {
```

Comment:
I'd name this something more explicit to delineate it from the data holders... try currentWeatherGenerator maybe which would be clear about its use.  Same as weekly weather below.